src/Domain/Aggregate/Building.php

```diff
@@ -34,22 +34,30 @@ public static function new(string $name) : self
         return $self;
     }
 
-    public function checkInUser(string $username)
+    public function checkInUser(string $username) : void
     {
-        // @TODO to be implemented
+        $this->recordThat(DomainEvent\UserCheckedIn::toBuilding(
+            $this->uuid,
+            $username
+        ));
     }
 
-    public function checkOutUser(string $username)
+    public function checkOutUser(string $username) : void
     {
         // @TODO to be implemented
     }
 
-    public function whenNewBuildingWasRegistered(NewBuildingWasRegistered $event)
+    protected function whenNewBuildingWasRegistered(DomainEvent\NewBuildingWasRegistered $event) : void
     {
         $this->uuid = Uuid::fromString($event->aggregateId());
         $this->name = $event->name();
     }
 
+    protected function whenUserCheckedIn(DomainEvent\UserCheckedIn $event) : void
```

Comment:
While this method is indeed unused, it is required to define all `protected function when{$eventName}($event)`, because as the project grows, many events related to an aggregate may go forgotten (because of their rarity), and if the `when...` methods are not implemented, these could turn into potentially dangerous silent failures.

Prooph by default requires the `when...` methods to be all declared, so we're quite safe.